tabular/src/autogluon/tabular/trainer/abstract_trainer.py

```diff
@@ -1432,6 +1491,67 @@ def _get_dummy_stacker(self, level: int, model_levels: dict, use_orig_features=T
         dummy_stacker.initialize(num_classes=self.num_classes)
         return dummy_stacker
 
+    def _proxy_model_feature_prune(self, feature_prune_kwargs: dict, time_limit: float, layer_fit_time: float, level: int, features: List[str]) -> List[str]:
+        """
+        Uses the best LightGBM-based base learner of this layer to perform time-aware permutation feature importance based feature pruning.
+        If all LightGBM models fail, use the model that achieved the highest validation accuracy. Feature pruning gets the smaller of the
+        remaining layer time limit and k times (default=2) it took to fit the base learners of this layer as its resource. Note that feature pruning can
+        exit earlier based on arguments in feature_prune_kwargs. The method returns the list of feature names that survived the pruning procedure.
+
+        Parameters
+        ----------
+        feature_prune_kwargs : dict
+            Feature pruning kwarg arguments. Should contain arguments passed to FeatureSelector.select_features. One can optionally attach the following
+            additional kwargs that are consumed at this level: 'proxy_model_class' to tell this method not to prioritize LGB as the proxy model,
+            'feature_prune_time_limit' to manually specify how long we should perform the feature pruning procedure for, 'k' to specify how long we should
+            perform feature pruning for if 'feature_prune_time_limit' has not been set, and 'raise_exception' to signify that AutoGluon should throw
+            an exception if feature pruning errors out.
+        time_limit : float
+            Time limit left within the current stack layer in seconds. Feature pruning should never take more than this time.
+        layer_fit_time : float
+            How long it took to fit all the models in this layer once. Used to calculate how long to feature prune for.
+        level : int
+            Level of this stack layer.
+        features: List[str]
+            The list of feature names in the inputted dataset.
+
+        Returns
+        -------
+        candidate_features : List[str]
+            Feature names that survived the pruning procedure.
+        """
+        leaderboard = self.leaderboard()
+        leaderboard = leaderboard[~leaderboard['model'].str.contains('WeightedEnsemble')]
+        fit_models = leaderboard[(~leaderboard['score_val'].isna()) & (leaderboard['stack_level'] == level)]
+        if len(fit_models) == 0:
+            return features
+        best_fit_models = fit_models.loc[fit_models['score_val'] == fit_models['score_val'].max()]
+        proxy_model = self.load_model(best_fit_models.loc[best_fit_models['fit_time'].idxmin()]['model'])
+
+        if feature_prune_kwargs[proxy_model.name].pop('proxy_model_class', "LGB") == "LGB":
```

Comment:
nit: `feature_prune_kwargs[proxy_model.name]` used 5 more times below in this method. Consider extracting it into a variable to improve readability.